src/handle-data.ts

```diff
@@ -85,6 +85,15 @@ export function getTotalPosition(layout: TableLayout) {
   return 'noTotals';
 }
 
+export function getProperties(reference: string, data: any) {
+  const steps: string[] = reference.split('.');
+  let value = data;
+  for (let i = 0; i < steps.length; ++i) {
+    if (value[steps[i]] !== null) value = value[steps[i]];
+  }
+  return value;
+}
```

Comment:
I meant that you don't need this function at all. It should do the same thing as optional chaining. It was not a part of the language when this function was written, not needed anymore

Reply:
uh... I don't know why I was thinking about having this func all the time... hadn't even thought about that. 
Removed it.